in_toto/rulelib_test.go

```diff
@@ -0,0 +1,80 @@
+package in_toto
+
+import (
+  "testing"
+  )
+
+func TestUnpackValidRules(t *testing.T) {
+
+  // A list of valid rules (lists)
+  // Each will be passed to rulelib.UnpackRule below
+  rules := [][]string {
+    []string{"CREATE", "foo"},
+    []string{"DELETE", "foo"},
+    []string{"MODIFY", "foo"},
+    []string{"ALLOW", "foo"},
+    []string{"DISALLOW", "foo"},
+    []string {"MATCH", "foo", "IN", "source-path", "WITH", "PRODUCTS", "IN",
+        "dest-path", "FROM", "step-name"},
+    []string {"MATCH", "foo", "IN", "source-path", "WITH", "MATERIALS",
+        "FROM", "step-name"},
+    []string {"MATCH", "foo", "WITH", "PRODUCTS", "IN", "dest-path",
+        "FROM", "step-name"},
+    []string {"MATCH", "foo", "WITH", "MATERIALS", "FROM", "step-name"},
+  }
+
+  // These are the expected results from rulelib.UnpackRule for above rules
+  // (associated by index)
+  expectedRuleMaps := []map[string]string {
+     map[string]string{"type": "create", "pattern": "foo"},
+     map[string]string{"type": "delete", "pattern": "foo"},
+     map[string]string{"type": "modify", "pattern": "foo"},
+     map[string]string{"type": "allow", "pattern": "foo"},
+     map[string]string{"type": "disallow", "pattern": "foo"},
+    map[string]string{"type": "match", "pattern": "foo",
+        "srcPrefix": "source-path", "dstPrefix": "dest-path",
+        "dstType": "products" , "dstName": "step-name"},
+    map[string]string{"type": "match", "pattern": "foo",
+        "srcPrefix": "source-path", "dstPrefix": "",
+        "dstType": "materials" , "dstName": "step-name"},
+    map[string]string{"type": "match", "pattern": "foo",
+        "srcPrefix": "", "dstPrefix": "dest-path",
+        "dstType": "products" , "dstName": "step-name"},
+    map[string]string{"type": "match", "pattern": "foo",
+        "srcPrefix": "", "dstPrefix": "",
+        "dstType": "materials" , "dstName": "step-name"},
+  }
+
+  for i := 0; i < len(rules); i++ {
```

Comment:
Use "for statements with `range` clause" perhaps? https://golang.org/ref/spec#For_statements

Reply:
Oh yes, makes perfect sense. Fixed in 20f9e1e9f6dbe07fa0aeca68e4d320d3da5b7992